lib/matplotlib/scale.py

```diff
@@ -103,13 +104,61 @@ def limit_range_for_scale(self, vmin, vmax, minpos):
         return vmin, vmax
 
 
+def _make_axis_parameter_optional(init_func):
+    """
+    Decorator to allow leaving out the *axis* parameter in scale constructors.
+
+    This decorator ensures backward compatibility for scale classes that
+    previously required an *axis* parameter. It allows constructors to be
+    callerd with or without the *axis* parameter.
+
+    For simplicity, this does not handle the case when *axis*
+    is passed as a keyword. However,
+    scanning GitHub, there's no evidence that that is used anywhere.
+
+    Parameters
+    ----------
+    init_func : callable
+        The original __init__ method of a scale class.
+
+    Returns
+    -------
+    callable
+        A wrapped version of *init_func* that handles the optional *axis*.
+
+    Notes
+    -----
+    If the wrapped constructor defines *axis* as its first argument, the
+    parameter is preserved when present. Otherwise, the value `None` is injected
+    as the first argument.
+
+    Examples
+    --------
+    >>> from matplotlib.scale import ScaleBase
+    >>> class CustomScale(ScaleBase):
+    ...     @_make_axis_parameter_optional
+    ...     def __init__(self, axis, custom_param=1):
+    ...         self.custom_param = custom_param
+    """
+    @wraps(init_func)
+    def wrapper(self, *args, **kwargs):
+        if args and isinstance(args[0], mpl.axis.Axis):
+            return init_func(self, *args, **kwargs)
+        else:
+            # Remove 'axis' from kwargs to avoid double assignment
+            kwargs.pop('axis', None)
+            return init_func(self, None, *args, **kwargs)
```

Comment:
Shouldn't the keyword-argument version be passed as the `axis` if it's there?
```suggestion
            # Remove 'axis' from kwargs to avoid double assignment
            return init_func(self, kwargs.pop('axis', None), *args, **kwargs)
```